### ISSUE TITLE
Resource context is not cancelled when apply aborted

### ISSUE THREAD

**Issue by cludden** [hashicorp/terraform-plugin-framework#781]
### Module version

```
github.com/hashicorp/terraform-plugin-framework v1.3.1
```

### Relevant provider source code

```go
package provider

import (
	"context"
	"time"

	"github.com/google/uuid"
	"github.com/hashicorp/terraform-plugin-framework/resource"
	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
	"github.com/hashicorp/terraform-plugin-framework/types"
)

// NewExampleResource constructor
func NewExampleResource() resource.Resource {
	return &ExampleResource{}
}

// ExampleResource defines the resource implementation.
type ExampleResource struct{}

// ExampleResourceModel describes the resource data model.
type ExampleResourceModel struct {
	Id types.String `tfsdk:"id"`
}

// Metadata returns resource metadata
func (r *ExampleResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
	resp.TypeName = req.ProviderTypeName + "_example"
}

// Schema describes an ExampleResource value
func (r *ExampleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
	resp.Schema = schema.Schema{
		Description: "example resource",

		Attributes: map[string]schema.Attribute{
			"id": schema.StringAttribute{
				Computed:            true,
				MarkdownDescription: "resource id",
				PlanModifiers: []planmodifier.String{
					stringplanmodifier.UseStateForUnknown(),
				},
			},
		},
	}
}

// Configure initializes an ExampleResource value
func (r *ExampleResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
}

// Create a workload
func (r *ExampleResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
	var m ExampleResourceModel
	if resp.Diagnostics.Append(req.Plan.Get(ctx, &m)...); resp.Diagnostics.HasError() {
		return
	}

	select {
	case <-time.After(time.Minute * 3):
	case <-ctx.Done():
		resp.Diagnostics.AddError("operation cancelled", "")
		return
	}

	m.Id = types.StringValue(uuid.NewString())
	resp.Diagnostics.Append(resp.State.Set(ctx, &m)...)
}

// Read a workload
func (r *ExampleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
}

// Update a workload
func (r *ExampleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
}

// Delete a workload
func (r *ExampleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
}
```

### Terraform Configuration Files

```hcl
resource "test_example" "this" {}
```

### Expected Behavior
Expect resource creation to detect cancelled context, log error, and return early.

### Actual Behavior
Resource finishes creation.

```shell
$ terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # f8n_example.this will be created
  + resource "test_example" "this" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

test_example.this: Creating...
test_example.this: Still creating... [10s elapsed]
^CStopping operation...

Interrupt received.
Please wait for Terraform to exit or data loss may occur.
Gracefully shutting down...

test_example.this: Still creating... [20s elapsed]
test_example.this: Still creating... [30s elapsed]
test_example.this: Still creating... [40s elapsed]
test_example.this: Still creating... [50s elapsed]
test_example.this: Still creating... [1m0s elapsed]
test_example.this: Still creating... [1m10s elapsed]
test_example.this: Still creating... [1m20s elapsed]
test_example.this: Still creating... [1m30s elapsed]
test_example.this: Still creating... [1m40s elapsed]
test_example.this: Still creating... [1m50s elapsed]
test_example.this: Still creating... [2m0s elapsed]
test_example.this: Still creating... [2m10s elapsed]
test_example.this: Still creating... [2m20s elapsed]
test_example.this: Still creating... [2m30s elapsed]
test_example.this: Still creating... [2m40s elapsed]
test_example.this: Still creating... [2m50s elapsed]
test_example.this: Creation complete after 3m0s [id=5285e1fb-3d00-401b-8cd3-ffc1b58777f7]
╷
│ Error: execution halted
│
│
╵
╷
│ Error: execution halted
│
│
╵
```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`
3. `ctrl+c` during resource creation

**Comment by bflad** [comment]
Hi @cludden 👋 Thank you so much for reporting this and the nice reproduction case. @austinvalle and I investigated this further and found out that this was an upstream issue with the provider server handling for any protocol version 6 provider server. We verified the fix, released the dependency, and will close this issue out as part of updating the dependency in this Go module.

If you need the fix quickest, feel free to pin your framework dependency to the commit SHA containing the fix or otherwise it will release with v1.3.2 (if we do cut a bug fix release next week) or v1.4.0. 👍

**Comment by chrismarget-j** [comment]
Thank you @cludden, @blfad and @austinvalle!

I'd noticed this issue as well, am happy to know it's already handled.
